reference/api/keys.md

```diff
@@ -1,30 +1,313 @@
+
+## Get one key
+
+<RouteHighlighter method="GET" route="/keys/:key"/>
 
-[More information about the keys and their rights](/reference/features/authentication.md).
+Get information on the specified key. Attempting to use this endpoint with a non-existent or deleted key will result in [an error](http://docs.meilisearch.com/errors#api_key_not_found).
 
-## Get keys
+#### Path variables
+
+| Variable        | Description                                                                  |
+| --------------- | ---------------------------------------------------------------------------- |
+| **key**         | Alphanumeric value generated by MeiliSearch on [key creation](#create-a-key) |
+
+### Example
+
+<CodeSamples id="get_one_key_1" />
+
+#### Response: `200 Ok`
+
+```json
+{
+  "description": "Add documents: Products API key",
+  "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+  "actions": [
+    "documents.add"
+  ],
+  "indexes": [
+    "products"
+  ],
+  "expiresAt": "2021-11-13T00:00:00Z",
+  "createdAt": "2021-11-12T10:00:00Z",
+  "updatedAt": "2021-11-12T10:00:00Z"
+}
+```
+
+For an explanation of these fields, see the [get all keys](#response-body) endpoint.
+
+## Get all keys
 
 <RouteHighlighter method="GET" route="/keys"/>
 
-Get the **private** and **public** [key](/reference/features/authentication.md).
+List all existing API keys. Expired keys are included in the response, but deleted keys are not.
+
+### Response body
+
+#### `description`
+
+**Type:** string
+**Modifiable:** yes
+
+A description for the key. Default value is `null`.
+
+#### `key`
+
+**Type:** string
+**Modifiable:** no
+
+An alphanumeric key value generated by MeiliSearch on API key creation. Used for authorization when [making calls to a protected MeiliSearch instance](/reference/features/authentication.md#communicating-with-a-protected-instance).
+
+This value is also used as the `:key` path variable to [update](#update-a-key), [delete](#delete-a-key), or [get information on](#get-one-key) a specific key.
+
+**Example**:
+
+```bash
+curl \
+  -X POST 'http://localhost:7700/indexes/products/documents' \
+  -H 'Authorization: Bearer d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4'
+  --data-binary '{
+    "id": 1,
+    "title": "Product 1",
+    "description": "Our first product!"
+    "price": 42
+  }'
+```
+
+#### `actions`
+
+**Type:** array
+**Modifiable:** yes
+
+A list of API actions permitted for the key. `["*"]` for all actions.
+
+The `*` character can also be used as a wildcard to get all permissions for a route, e.g. `documents.*` to authorize access to all document-related endpoints.
+
+#### `indexes`
+
+**Type:** array
+**Modifiable:** yes
+
+A list of indexes permitted for the key. `["*"]` for all indexes.
+
+#### `expiresAt`
+
+**Type:** string
+**Modifiable:** yes
+
+Date and time when the key will expire, represented in **ISO-8601** format. `null` if the key never expires.
```

Comment:
@maryamsulemani97 has been writing `ISO 8601` (without the dash).  I have my doubts about the usefulness of mentioning ISO formats by their code, but I think we should make sure we're consistent in how we call it.

This comment also applies to the two fields below, `createdAt` and `updatedAt`.

Reply:
I used ISO 8601 instead of `ISO 8601`.

Reply:
Yeap, sorry, I accidentally used backticks instead of quotes for some reason.

Also, [it seems ISO also prefer not to use the hyphen](https://www.iso.org/iso-8601-date-and-time-format.html). I think it makes sense to follow their usage and format references to ISO standards as @maryamsulemani97 has been doing.

```suggestion
Date and time when the key will expire, represented in **ISO 8601** format. `null` if the key never expires.
```